### PR TITLE
feat: run expo prebuild, install detox, jest, expo config for detox

### PR DESCRIPTION
**Pull request by edgarkhanzadian**
Setup e2e tests in the mobile app

## Summary by CodeRabbit

- **New Features**
  - Introduced Detox testing framework configuration for mobile apps.
  - Added ProGuard rules for Android app development.
  - Updated Android and iOS project settings and configurations for React Native and Expo integration.

- **Bug Fixes**
  - Updated `.gitignore` files to exclude Detox artifacts and other irrelevant files from version control.

- **Documentation**
  - Added `e2e-testing.md` with instructions for running end-to-end tests using Detox.

- **Chores**
  - Updated Gradle and Xcode project settings for improved build and dependency management.

### CODE REVIEW

**Review by fbwoolf — APPROVED** [review]

File: apps/mobile/android/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    reactNativeDir = new File(["node", "--print", "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim()).getParentFile().getAbsoluteFile()
+    hermesCommand = new File(["node", "--print", "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim()).getParentFile().getAbsolutePath() + "/sdks/hermesc/%OS-BIN%/hermesc"
+    codegenDir = new File(["node", "--print", "require.resolve('@react-native/codegen/package.json', { paths: [require.resolve('react-native/package.json')] })"].execute(null, rootDir).text.trim()).getParentFile().getAbsoluteFile()
+
+    // Use Expo CLI to bundle the app, this ensures the Metro config
+    // works correctly with Expo projects.
+    cliFile = new File(["node", "--print", "require.resolve('@expo/cli', { paths: [require.resolve('expo/package.json')] })"].execute(null, rootDir).text.trim())
+    bundleCommand = "export:embed"
+
+    /* Folders */
```
Comment (pete-watters) [inline apps/mobile/android/app/build.gradle:22]: Could we remove the commented out code?

**Review by pete-watters — APPROVED** [review]
Nice work! 🚀  LGTM. 

The rabbit found some things in `package.json` that we could want to address? Maybe we know better though

File: apps/mobile/android/app/src/debug/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
```
Comment (kyranjamie) [inline apps/mobile/android/app/src/debug/AndroidManifest.xml:4]: 2 vs 4 space tabs as above. Should we use https://github.com/prettier/plugin-xml

**Review by kyranjamie — COMMENTED** [review]
How do I run these tests?

Edit: I've been through detox docs, and running into a build error

```bash
❯ pnpm detox build --configuration ios.release
xcodebuild -workspace ios/leatherwalletmobile.xcworkspace -scheme leatherwalletmobile -configuration Release -sdk iphonesimulator -arch x86_64 -derivedDataPath ios/build
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -workspace ios/leatherwalletmobile.xcworkspace -scheme leatherwalletmobile -configuration Release -sdk iphonesimulator -arch x86_64 -derivedDataPath ios/build

User defaults from command line:
    IDEDerivedDataPathOverride = /Users/kyranjamie/dev/leather/mono/apps/mobile/ios/build
    IDEPackageSupportUseBuiltinSCM = YES

Build settings from command line:
    ARCHS = x86_64
    SDKROOT = iphonesimulator17.5

2024-06-04 12:02:22.664 xcodebuild[32614:18894755] Writing error result bundle to /var/folders/n2/ds19zkvd5q52dtfnyyb_qdbr0000gn/T/ResultBundle_2024-04-06_12-02-0022.xcresult
xcodebuild: error: Found no destinations for the scheme 'leatherwalletmobile' and action build.


Important: 'detox build' is a convenience shortcut for calling your own build command, as provided in the config file.
Failures in this build command are not the responsibility of Detox. You are responsible for maintaining this command.

Command failed: xcodebuild -workspace ios/leatherwalletmobile.xcworkspace -scheme leatherwalletmobile -configuration Release -sdk iphonesimulator -arch x86_64 -derivedDataPath ios/build
```

**Pull request merged**